gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/util/BloodPressureValidatorUtilTest.java

```diff
@@ -0,0 +1,132 @@
+    private static final List<String> DIASTOLIC_2 = Arrays.asList("271650006", "406508013");
+    private static final List<String> DIASTOLIC_3 = Arrays.asList("400975005", "1780183016");
+    private static final List<String> DIASTOLIC_4 = Arrays.asList("407555005", "2159156012");
+    private static final List<String> DIASTOLIC_5 = Arrays.asList("407557002", "2159158013");
+
+    private static final String INVALID_CODE = "123456789";
+
+    @ParameterizedTest(name = "validBloodPressureTriple")
+    @MethodSource("validBloodPressureTriples")
+    public void validBloodPressureTripleReturnsTrue(String header, String systolic, String diastolic) {
+        assertThat(BloodPressureValidatorUtil.validateBloodPressureTriple(header, systolic, diastolic)).isTrue();
+    }
+
+    @Test
+    public void invalidBloodPressureTripleReturnsFalse() {
+        assertThat(BloodPressureValidatorUtil.validateBloodPressureTriple("123456789", "987654321",
+            "192837465")).isFalse();
+    }
+
+    @ParameterizedTest(name = "validSystolicBloodPressure")
+    @MethodSource("validSystolicBloodPressures")
+    public void validSystolicBloodPressureCodeReturnsTrue(String systolicCode) {
+        assertThat(BloodPressureValidatorUtil.isSystolicBloodPressure(systolicCode)).isTrue();
+    }
+
+    @Test
+    public void invalidSystolicBloodPressureCodeReturnsFalse() {
+        assertThat(BloodPressureValidatorUtil.isSystolicBloodPressure(INVALID_CODE)).isFalse();
+    }
+
+    @ParameterizedTest(name = "validDiastolicBloodPressure")
+    @MethodSource("validDiastolicBloodPressures")
+    public void validDiastolicBloodPressureCodeReturnsTrue(String diastolicCode) {
+        assertThat(BloodPressureValidatorUtil.isDiastolicBloodPressure(diastolicCode)).isTrue();
+    }
+
+    @Test
+    public void invalidDiastolicBloodPressureCodeReturnsFalse() {
+        assertThat(BloodPressureValidatorUtil.isDiastolicBloodPressure(INVALID_CODE)).isFalse();
+    }
+
+    private static Stream<Arguments> validBloodPressureTriples() {
+        return Stream.of(
+                generateBloodPressureTriples(HEADER_1, SYSTOLIC_1, DIASTOLIC_1),
+                generateBloodPressureTriples(HEADER_1, SYSTOLIC_2, DIASTOLIC_2),
+                generateBloodPressureTriples(HEADER_2, SYSTOLIC_1, DIASTOLIC_1),
+                generateBloodPressureTriples(HEADER_2, SYSTOLIC_2, DIASTOLIC_2),
+                generateBloodPressureTriples(HEADER_3, SYSTOLIC_1, DIASTOLIC_1),
+                generateBloodPressureTriples(HEADER_3, SYSTOLIC_2, DIASTOLIC_2),
+                generateBloodPressureTriples(HEADER_4, SYSTOLIC_3, DIASTOLIC_3),
+                generateBloodPressureTriples(HEADER_5, SYSTOLIC_4, DIASTOLIC_4),
+                generateBloodPressureTriples(HEADER_6, SYSTOLIC_5, DIASTOLIC_5))
+            .flatMap(Collection::stream)
+            .collect(Collectors.toList()).stream();
```

Comment:
I think this could be replaced with
```suggestion
            .flatMap(Collection::stream);
```